client/client.go

```diff
@@ -20,17 +21,12 @@ type Client struct {
 	responseQueue   *amqp.Queue
 }
 
-// NewBroker creates a new instance of AMQP connection.
-func NewBroker(amqpURL string) (*amqp.Connection, error) {
-	return amqp.Dial(amqpURL)
-}
-
 // NewClient creates a new instance of Client, responsible for making remote calls.
-func NewClient(conn *amqp.Connection, serviceName string, defaultTTL int64) (*Client, error) {
-	ch, err := conn.Channel()
+func NewClient(b *broker.Broker, serviceName string, defaultTTL int64) (*Client, error) {
+	ch, err := b.Conn.Channel()
```

Comment:
Conn could be internal and non related with AMQP, so if we want a new broker Redis bases it wont break the code.

It would be nice to have broker.Broker as interface and a amqp.Broker as the real implementation.